### ISSUE TITLE
Can't reproduce the results!!!

### ISSUE THREAD

**Issue by fighting-liu** [TuSimple/mx-maskrcnn#59]
Hi, firstly, thanks for your contribution, but i can't reproduce the results as you mentiond 31.3 for AP and 55.5 for AP_50%   on val dataset of Cityscape.  I use the recently updated code and 4 GPUs follow the **train_alternate_mask_fpn.py** script, everything works fine and got 26.4 for AP and 50.0 for AP_50%.

**Comment by winstywang** [comment]
Have you recently pull the latest master to fix the mask flip bug?

**Comment by fighting-liu** [comment]
Yes, i've  used the  latest master.

**Comment by 501177639** [comment]
##################################################
what           :             AP         AP_50%
##################################################
person         :          0.292          0.601
rider          :          0.245          0.643
car            :          0.483          0.728
truck          :          0.212          0.325
bus            :          0.369          0.516
train          :          0.200          0.369
motorcycle     :          0.164          0.460
bicycle        :          0.183          0.498
--------------------------------------------------
average        :          0.269          0.518

Hi， I got this result.

**Comment by Zehaos** [comment]
Hi, @fighting-liu @501177639 
I'm working on this, I found a mistake made in code refinement. I will make a PR to fix it soon.

**Comment by ypflll** [comment]
My result even worse:

what           :             AP         AP_50%
##################################################
person         :          0.275          0.578
rider          :          0.237          0.638
car            :          0.485          0.731
truck          :          0.192          0.305
bus            :          0.369          0.517
train          :          0.128          0.340
motorcycle     :          0.132          0.381
bicycle        :          0.166          0.464
--------------------------------------------------
average        :          0.248          0.494

**Comment by Zehaos** [comment]
I just fix it. Please pull the newest version. Refer to PR #66.

I got the result below with [factor=0.03](https://github.com/TuSimple/mx-maskrcnn/blob/master/rcnn/PY_OP/mask_output.py#L13) and default configuration.  
I'm now trying factor=0.02, it seems better than 0.03. So I set factor=0.02 in the PR.
```
##################################################
what           :             AP         AP_50%
##################################################
person         :          0.293          0.611
rider          :          0.228          0.614
car            :          0.483          0.720
truck          :          0.236          0.355
bus            :          0.484          0.667
train          :          0.366          0.581
motorcycle     :          0.164          0.402
bicycle        :          0.182          0.497
--------------------------------------------------
average        :          0.305          0.556
```

**Comment by fighting-liu** [comment]
Ok, i will try it again and report my results here later!

**Comment by jiang1st** [comment]
@Zehaos I got this result with the latest PR (factor = 0.02):

##################################################
what           :             AP         AP_50%
##################################################
person         :          0.288          0.602
rider          :          0.227          0.620
car            :          0.475          0.708
truck          :          0.251          0.392
bus            :          0.427          0.600
train          :          0.281          0.559
motorcycle     :          0.172          0.417
bicycle        :          0.177          0.506
--------------------------------------------------
average        :          0.287          0.550​​

I used 8 GPUs so I doubled the base learning rate to 0.008. Could that be a problem?

**Comment by ypflll** [comment]
@jiang1st  I got the same result with you.
@Zehaos Why do you think factor 0.02 is better? Seems that we get a little lower mAP.

**Comment by Zehaos** [comment]
Hey, guys, The original code has a small performance gap with our pretrained model.( ~ 1% drop on AP100, no effect on AP50). With commit #82, the code now should reproduce our result on cityscapes dataset.

**Comment by jiang1st** [comment]
Hi @Zehaos, thank you for your prompt reply. What is "factor" used for?  To control the magnitude of gradients coming from the mask branch?

**Comment by Zehaos** [comment]
@jiang1st Exactly.

**Comment by jiang1st** [comment]
@Zehaos Is it a common practice to adjust this parameter? I didn't see it anywhere in the original paper. Besides, why did you set it to 0.03125 ? Thanks.

**Comment by Zehaos** [comment]
@jiang1st It is not mentioned in the paper. In my opinions, it is important to re-weight the additional branch. We set it to 0.25 originally, 0.03125 = 0.25/8. (because of code refinements)

**Comment by jiang1st** [comment]
@Zehaos Hi Zehao, I got this result. Possibly because I doubled the base_lr with 8 GPUs? I also notice you are using base_lr = 0.004 with 4 GPUs (Original paper use 0.01 with 8 GPUs). Is there any suggestion on adjusting base_lr and actual batch_size during training process? Thanks a lot.
##################################################
what           :             AP         AP_50%
##################################################
person         :          0.324          0.631
rider          :          0.262          0.656
car            :          0.493          0.722
truck          :          0.235          0.379
bus            :          0.454          0.635
train          :          0.257          0.522
motorcycle     :          0.161          0.404
bicycle        :          0.200          0.527
--------------------------------------------------
average        :          0.298          0.559

**Comment by Zehaos** [comment]
@jiang1st There is class imbalance problem( AP of truck, bus and train is unstable). According to AP of person and car, it is normal of your result. 
In detection task, people tend to use larger batch size(Faster:2, FPN:16, MegDet:256). We set a batch size much smaller than the original paper, maybe a larger batch size can make some improvement.

**Comment by jiang1st** [comment]
@Zehaos What's your strategy to adjust the base learning rate (based on the batch size)? Can you offer me some advice? Thank you.

**Comment by huangzehao** [comment]
We have reproduced the reported results by our latest codes.  Please pull the newest version, thanks.

##################################################
what : AP AP_50%
##################################################
person : 0.326 0.637
rider : 0.267 0.656
car : 0.494 0.727
truck : 0.285 0.401
bus : 0.469 0.644
train : 0.303 0.562
motorcycle : 0.181 0.428
bicycle : 0.202 0.532
-------------------------------------------------------------------------------------
average: 0.316 0.573
